library/pkwrite.c

```diff
@@ -53,6 +53,28 @@
 #endif
 #include "mbedtls/platform.h"
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO) && defined(MBEDTLS_ECP_LIGHT)
+static psa_ecc_family_t pk_get_opaque_ec_family(psa_key_attributes_t *key_attrs)
+{
+    psa_key_type_t key_type = PSA_KEY_TYPE_NONE;
+
+    key_type = psa_get_key_type(key_attrs);
+    if (!PSA_KEY_TYPE_IS_ECC(key_type)) {
+        return 0;
+    }
+
+    return PSA_KEY_TYPE_ECC_GET_FAMILY(psa_get_key_type(key_attrs));
+}
+
+static mbedtls_ecp_group_id pk_get_opaque_ec_group_id(psa_key_attributes_t *key_attrs)
+{
+    psa_ecc_family_t ec_family = pk_get_opaque_ec_family(key_attrs);
```

Comment:
`ec_family` is not checked for 0. PSA specification states that:

> 0x00 – 0x7f
>ECC family identifiers defined by this standard. Unallocated values in this range are reserved for future use.

So a clash is possible.

Reply:
It was my intent that 0 would never be a valid EC family (the same way it's explicitly reserved e.g. in `psa_key_type_t`), but I somehow didn't end up writing it explicitly. [I proposed to add this statement to the PSA specification](https://github.com/ARM-software/psa-api/issues/67).

Reply:
You can assume that 0 is not valid. It's not valid now and PSA won't make it valid in the future.